xy/line_centroid_test.go

```diff
@@ -62,30 +64,54 @@ var lineTestData = []struct {
 
 func TestLineGetCentroidLines(t *testing.T) {
 	for i, tc := range lineTestData {
-		centroid := xy.LinesCentroid(tc.lines[0], tc.lines[1:]...)
+		verifyLineCentroid(t, i, tc)
+		verifyMultiLineCentroid(t, i , tc)
+		verifyLinearRingsCentroid(t, i , tc)
+	}
 
-		if !reflect.DeepEqual(tc.lineCentroid, centroid) {
-			t.Errorf("Test '%v' failed: expected centroid for polygon array to be\n%v but was \n%v", i+1, tc.lineCentroid, centroid)
-		}
+}
+func verifyLineCentroid(t *testing.T, i int, tc lineDataType) {
+	centroid := xy.LinesCentroid(tc.lines[0], tc.lines[1:]...)
 
-		coords := []float64{}
-		ends := []int{}
-		for _, p := range tc.lines {
-			coords = append(coords, p.FlatCoords()...)
-			ends = append(ends, len(coords))
-		}
+	if !reflect.DeepEqual(tc.lineCentroid, centroid) {
+		t.Errorf("Test '%v' failed: expected centroid for polygon array to be\n%v but was \n%v", i + 1, tc.lineCentroid, centroid)
```

Comment:
Test indexes make tracking down the exact test that failed hard for humans. Better to use a human-readable text field to the test case (i.e. add `struct TestCase { desc string }` and print `desc` instead of the test index).